### PR TITLE
feat: Add ability to style DatePickerFlyoutPresenter

### PR DESCRIPTION
**Pull request by kazo0**
GitHub Issue (If applicable): https://github.com/unoplatform/Uno.Themes/issues/470

## PR Type

What kind of change does this PR introduce?

- Feature

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs [hunk 1]
```diff
@@ -27,14 +27,30 @@ public bool UseNativeStyle
 			set => SetValue(UseNativeStyleProperty, value);
 		}
 
+		// FlyoutPresenterStyle is an Uno-only property to allow the styling of the DatePicker's FlyoutPresenter.
```
Comment (ajpinedam) [inline src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs:30]: Should this be an XML Document?

File: src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs [hunk 2]
```diff
@@ -27,14 +27,30 @@ public bool UseNativeStyle
...
+		public Style FlyoutPresenterStyle
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs:31]: Adding the UnoSpecific attribute would be useful

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePickerFlyout.Properties.cs [hunk 3]
```diff
@@ -128,5 +128,20 @@ public DateTimeOffset MaxYear
...
 
+		public Style DatePickerFlyoutPresenterStyle
+		{
+			get { return (Style)this.GetValue(DatePickerFlyoutPresenterStyleProperty); }
+			set { this.SetValue(DatePickerFlyoutPresenterStyleProperty, value); }
+		}
+
+		public static DependencyProperty DatePickerFlyoutPresenterStyleProperty { get; } =
+			DependencyProperty.Register(
+				"DatePickerFlyoutPresenterStyle",
```
Comment (ajpinedam) [inline src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePickerFlyout.Properties.cs:139]: nit: `nameof(DatePickerFlyoutPresenterStyle)`

**Review by ajpinedam — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs [hunk 4]
```diff
@@ -67,6 +84,9 @@ void OnPicked(DatePickerFlyout snd, DatePickedEventArgs evt)
 					DateChanged?.Invoke(this, new DatePickerValueChangedEventArgs(evt.NewDate, evt.OldDate));
 				}
 			}
+
+			DatePickerFlyout CreateManagedDatePickerFlyout()
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs:88]: Add access modifier please :)
```suggestion
			private DatePickerFlyout CreateManagedDatePickerFlyout()
```

File: src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs (hunk 2 above)
Comment (kazo0) [reply src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs:31]: I don't see an UnoSpecificAttribute, I see a lot of "Uno specific" comments but no attribute

Comment (ajpinedam) [reply src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs:33]: @kazo0  I think that @jeromelaban refers to:

//TODO: Uno specific

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs:34]: there's this one: https://github.com/unoplatform/uno/blob/master/src/Uno.Foundation/UnoOnlyAttribute.cs

**Review by davidjohnoliver — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs (hunk 2 above)
Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs:34]: `//TODO: Uno specific` only makes sense in translated C++ code for workarounds that should ultimately be removed (hence the 'TODO' part)

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Flyout.cs:34]: Ah yes UnoOnly, apologies :)

**Pull request merged**